tests/Mado/QueryBundle/Component/Meta/JsonPathFinderTest.php

```diff
@@ -0,0 +1,466 @@
+        );
+
+        $this->pathFinder->setEntity("AppBundle\\Entity\\Root");
+
+        $this->assertEquals(
+            "foo.bar.fizz",
+            $this->pathFinder->getPathTo("AppBundle\\Entity\\Fizz")
+        );
+    }
+
+    public function testBuildPathBetweenTwoEntities()
+    {
+        $this->samepleJson = [
+            "GammaBundle\\Entity\\Item" => [
+                "relations" => [
+                    "items" => "AppBundle\\Entity\\Foo",
+                ]
+            ],
+            "AppBundle\\Entity\\Foo" => [
+                "relations" => [
+                    "item" => "ZarroBundle\\Entity\\Item",
+                ]
+            ],
+            "ZarroBundle\\Entity\\Item" => [
+                "relations" => [
+                    "family" => "AppBundle\\Entity\\Family",
+                ]
+            ],
+        ];
+
+        $this->mapper = $this
+            ->getMockBuilder('Mado\QueryBundle\Component\Meta\RelationDatamapper')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->mapper->expects($this->once())
+            ->method('getMap')
+            ->will($this->returnValue(
+                $this->samepleJson
+            ));
+
+        $this->pathFinder = new JsonPathFinder(
+            $this->mapper
+        );
+
+        $this->pathFinder->setQueryStartEntity("GammaBundle\\Entity\\Item");
+
+        $this->assertEquals(
+            "_embedded.items.item.family",
+            $this->pathFinder->getPathToEntity("AppBundle\\Entity\\Family")
+        );
+    }
+
+    /**
+     * @expectedException \Mado\QueryBundle\Component\Meta\Exceptions\UnreachablePathException
+     */
+    public function testThrowExceptionIfPathNotExists()
+    {
+        $this->samepleJson = [
+            "GammaBundle\\Entity\\Merenghe" => [
+                "relations" => [
+                    "items" => "AppBundle\\Entity\\Foo",
+                ]
+            ],
+            "AppBundle\\Entity\\Foo" => [
+                "relations" => [
+                    "item" => "ZarroBundle\\Entity\\Item",
+                ]
+            ],
+            "ZarroBundle\\Entity\\Item" => [
+                "relations" => [
+                    "family" => "AppBundle\\Entity\\Family",
+                ]
+            ],
+        ];
+
+        $this->mapper = $this
+            ->getMockBuilder('Mado\QueryBundle\Component\Meta\RelationDatamapper')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->mapper->expects($this->once())
+            ->method('getMap')
+            ->will($this->returnValue(
+                $this->samepleJson
+            ));
+
+        $this->pathFinder = new JsonPathFinder(
+            $this->mapper
+        );
+
+        $this->pathFinder->setQueryStartEntity("GammaBundle\\Entity\\Item");
+        $this->pathFinder->getPathToEntity("AppBundle\\Entity\\Family");
+    }
+
+    public function testCountNumberOfParentOfInnerEntity()
```

Comment:
It's not so clear this test, can you please fix some name?

Reply:
Ok. When the map of entities and relation is built, for each application entity (in the map here an entity is a "parent entity") we have "inner entities". Maybe "relationEntity" instead of "innerEntity" could be more clear?

```
'ParentEntity' => [
    'relations' => [
        'relation_name' => 'InnerEntity',
        'relation_name1' => 'InnerEntity1',
        'relation_name2' => 'InnerEntity2',
        'relation_name3' => 'InnerEntity3',
    ]
],
'ParentEntity1' => [
    'relations' => [
        'relation_name' => 'InnerEntity',
        'relation_name1' => 'InnerEntity1',
        'relation_name2' => 'InnerEntity2',
        'relation_name3' => 'InnerEntity3',
    ]
],
```

Reply:
Ok It's more clear I think